components/eam/src/physics/crm/physpkg.F90

```diff
@@ -1805,1229 +1180,654 @@ subroutine tphysac (ztodt,   cam_in,  &
+  real(r8),pointer :: snow_dp(:)                ! snow from ZM convection
+  real(r8),pointer :: prec_sh(:)                ! total precipitation from Hack convection
+  real(r8),pointer :: snow_sh(:)                ! snow from Hack convection
+
+  ! stratiform precipitation variables
+  real(r8),pointer :: prec_str(:)    ! sfc flux of precip from stratiform (m/s)
+  real(r8),pointer :: snow_str(:)     ! sfc flux of snow from stratiform   (m/s)
+  real(r8),pointer :: prec_pcw(:)     ! total precip from prognostic cloud scheme
+  real(r8),pointer :: snow_pcw(:)     ! snow from prognostic cloud scheme
+  real(r8),pointer :: prec_sed(:)     ! total precip from cloud sedimentation
+  real(r8),pointer :: snow_sed(:)     ! snow from cloud ice sedimentation
+  real(r8) :: sh_e_ed_ratio(pcols,pver)       ! shallow conv [ent/(ent+det)] ratio  
+
+  ! Local copies for substepping
+  real(r8) :: prec_pcw_macmic(pcols)
+  real(r8) :: snow_pcw_macmic(pcols)
+  real(r8) :: prec_sed_macmic(pcols)
+  real(r8) :: snow_sed_macmic(pcols)
+
+  ! energy checking variables
+  real(r8) :: zero(pcols)                    ! array of zeros
+  real(r8) :: rliq(pcols)                    ! vertical integral of liquid not yet in q(ixcldliq)
+  real(r8) :: rliq2(pcols)                   ! vertical integral of liquid from shallow scheme
+  real(r8) :: det_s  (pcols)                 ! vertical integral of detrained static energy from ice
+  real(r8) :: det_ice(pcols)                 ! vertical integral of detrained ice
+  real(r8) :: flx_cnd(pcols)
+  real(r8) :: flx_heat(pcols)
+  type(check_tracers_data):: tracerint             ! energy integrals and cummulative boundary fluxes
+  real(r8) :: zero_tracers(pcols,pcnst)
+
+  logical   :: lq(pcnst)
+
+  character(len=fieldname_len)   :: varname, vsuffix
+
+  !BSINGH - these were moved from zm_conv_intr because they are  used by aero_model_wetdep 
+  real(r8):: mu(pcols,pver) 
+  real(r8):: eu(pcols,pver)
+  real(r8):: du(pcols,pver)
+  real(r8):: md(pcols,pver)
+  real(r8):: ed(pcols,pver)
+  real(r8):: dp(pcols,pver)
+  real(r8):: dsubcld(pcols) ! wg layer thickness in mbs (between upper/lower interface).
+  integer :: jt(pcols)      ! wg layer thickness in mbs between lcl and maxi.    
+  integer :: maxg(pcols)    ! wg top  level index of deep cumulus convection.
+  integer :: ideep(pcols)   ! wg gathered values of maxi.
+  integer :: lengath        ! w holds position of gathered points vs longitude index
+
+  real(r8) :: ftem(pcols,pver)         ! tmp space
+  real(r8), pointer, dimension(:) :: static_ener_ac_2d ! Vertically integrated static energy
+  real(r8), pointer, dimension(:) :: water_vap_ac_2d   ! Vertically integrated water vapor
+  real(r8) :: CIDiff(pcols)            ! Difference in vertically integrated static energy
+
+  logical :: l_bc_energy_fix, l_dry_adj, l_tracer_aero, l_st_mac, l_st_mic, l_rad
+  
+  logical           :: use_MMF
```

Comment:
I noticed you took this flag out of `phys_register`, but left it in here. I assume that physpkg will now no longer work with `use_MMF=.false.`, so why even have this flag in here at all?

Reply:
good point, I'll remove it. We might be able to remove it entirely now that I think of it...